glslc/src/file_compiler.h

```diff
@@ -101,6 +105,23 @@ class FileCompiler {
   };
 
  private:
+  enum class OutputType {
+    SpirvBinary,  // A binary module, as defined by the SPIR-V specification.
+    SpirvAssemblyText,  // Assembly syntax defined by the SPIRV-Tools project.
+    PreprocessedText,   // Preprocessed source code.
+  };
+
+  // Compiles the given string a the specified shader kind and using stored
+  // options. On success, writes the result to the given output stream and
+  // returns true.  On failure, returns false, and possibly emit messages to
+  // the standard error stream.
+  template <typename CompilationResultType>
+  bool EmitCompiledResult(
+      const CompilationResultType& result, const std::string& input_file_name,
+      shaderc_util::string_piece error_file_name,
+      const std::unordered_set<std::string>& used_source_files,
+      std::ostream* out);
```

Comment:
Turns out, no since the method also accumulates warning and error counts.  That will be mentioned in the blurb.